File: AngularBlog.API/Models/Article.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AngularBlog.API.Models
{
    public class Article
    {
        public Article()
        {
            Comment = new HashSet<Comment>();
        }

        public int Id { get; set; }
        public string Title { get; set; }
        public string ContentSummary { get; set; }
        public string ContentMain { get; set; }
        public DateTime PublishDate { get; set; }
        public string Picture { get; set; }
        public int CategoryId { get; set; }
        public int ViewCount { get; set; }

        public virtual Category Category { get; set; }
        public virtual ICollection<Comment> Comment { get; set; }
    }
}

File: AngularBlog-UI/src/app/models/article.ts
import { Category } from "./category";

export class Article {
  id:number;
  title:string;
  contentMain:string;
  contentSummary:string;
  publishDate:Date;
  picture:string;
  viewCount:number;
  commentCount:number;
  category:Category;
  constructor(id: number, title: string,contentMain:string,contentSummary:string,publishDate:Date,picture:string,viewCount:number,commentCount:number,category:Category) {
    this.id = id;
    this.title = title;
    this.contentMain = contentMain;
    this.contentSummary = contentSummary;
    this.publishDate = publishDate;
    this.picture = picture;
    this.viewCount = viewCount;
    this.commentCount = commentCount;
    this.category = category;

  }
}
